pkg/client/dtclient/config_client.go

```diff
@@ -334,6 +336,9 @@ func isApplicationNotReadyYet(resp rest.Response, theApi api.API) bool {
 		isApplicationDetectionRuleException(resp, theApi) ||
 		strings.Contains(string(resp.Body), "Unknown application(s)")
 }
+func isNetworkZoneFeatureNotEnabledYet(resp rest.Response, theApi api.API) bool {
+	return strings.HasPrefix(theApi.ID, "network-zone") && (resp.Is4xxError())
```

Comment:
It makes it more prone to breaking, but should we maybe be more explicit than 4xx? E.g. the isApplicationNotReadyYet checks a specific response content, and thus is tied to the message but won't retry for any possibly unrelated user error.